### ISSUE TITLE
Mac OS Crash Loop after Toggling Proxy Setting

### ISSUE THREAD

**Issue by amosgroth** [Blockstream/green_qt#79]
### Description

Downloaded Green for the very first time, installed and opened it. Before setting everything up, I wanted to make sure communication goes through Tor. Jumped to the settings menu, activated the toggle for "Proxy", the application immediately crashed and is not usable ever since. Every time I want to restart the app, the Mac OS error dialogue shows up instead, telling me to send the crash report. Makes no difference, if I click on "send" or not, the dialogue closes after the click but the application won't start so I am stuck in this loop and cannot use the application at all.

### Version

1.1.6

### Steps to reproduce

1. Install Green
2. Open Green
3. Go to settings
4. Toggle "Proxy" (Crash)
5. Reopen Green

### Expected behaviour

* The proxy toggle should not crash the application.
* The application should reopen properly again after a crash.
* There should be a clear and human understandable error message in case of a problem/crash.

### Actual behaviour

* The proxy toggle (activation) crashes the application.
* The application doesn't reopen at all after.
* There is no clear human error message.

### Screenshots

![Capture d’écran 2023-02-01 à 07 21 23](https://user-images.githubusercontent.com/2759719/216033550-a70b6028-ef7f-4a93-a7d4-2ff7dba1c6b8.png)

### Device or machine

![Capture d’écran 2023-02-01 à 07 44 42](https://user-images.githubusercontent.com/2759719/216033727-5edcf307-88e5-44d2-9a88-d2739d464a8a.png)

[macOS Crash Report.log](https://github.com/Blockstream/green_qt/files/10556585/macOS.Crash.Report.log)

**Comment by amosgroth** [comment]
So after waiting for more than one month for a reply here, I suppose Blockstream is not willing or unable to help. Not very professional, to say the least. I'll stick to my BitBox02 and Shiftcrypto. Their device and software runs without problems at all and their support is fast, friendly, reliable and they care about their software...

**Comment by domegabri** [comment]
Hello @amosgroth, your report has been noted. We are on the middle of a large refactor that will address many issues, including this one. We should have a release within the next week or so. Sorry for the inconvenience.

**Comment by domegabri** [comment]
Hello @amosgroth , this should be fixed in the latest release.

**Comment by amosgroth** [comment]
Update: Deleting all data & reinstalling doesn't help. Even with standard settings, the app freezes right after start while trying to connect to Blockstream servers...

**Comment by domegabri** [comment]
Is the node that you are connecting to running in LAN? Do you have other optiona enabled like spv?

**Comment by amosgroth** [comment]
Yes, it's in the LAN. In the UI I don't see clearly what is happening, if I am connected or not, what the problem is etc. But in the meantime I tested the Linux and the Android App. They don't crash but they don't recognize the device, neither via USB nor BT. I am giving up...

**Comment by amosgroth** [comment]
I repeat:

- The Mac application crashes/freezes as described, no matter what configuration, also with Blockstream server standard settings (Tor activated)
- There is no clear indication for the user to see what's happening, if the application is connected or not or if there is any error
- Also it seems like you need to restart the application for changed server settings to take effect, which is not communicated nor plausible and pretty old school
- Activating Tor and connecting to a LAN server at the same time makes sense to me, the same feature works like a charm with Sparrow (described here: https://sparrowwallet.com/docs/connect-node.html#local-setup)

As said earlier, Linux and Android app don't work neither, though with a different problem (not being able to find and connect to the device).

**Comment by borgbyte** [comment]
@amosgroth can you check 1.2.0 and let us know how it goes.

**Comment by borgbyte** [comment]
@amosgroth try https://github.com/Blockstream/green_qt/issues/82#issuecomment-1555306466

**Comment by amosgroth** [comment]
> @amosgroth try [#82 (comment)](https://github.com/Blockstream/green_qt/issues/82#issuecomment-1555306466)

OK, that fix works.

* Still I have to restart the app after changing the server to my personal one. The app doesn't tell me so I have to guess.
* Also to me it's still not clear wether I am connected or not. No tooltip on the "connection" icon, no message from the application. I suppose I am connected when the wheel stops spinning. After a few minutes it seems to lose connection (icon disappears?). Tested aside with Sparrow where I get feedback and stay connected.
* Next problem: No connection of the device possible, neither with USB nor with BLE. Terminal (lsusb) tells me: `Bus 003 Device 005: ID 1a86:55d4 QinHeng Electronics USB Single Serial` so it is properly connected. The application doesn't show or tell me anything about it.

**Comment by amosgroth** [comment]
> @amosgroth try [#82 (comment)](https://github.com/Blockstream/green_qt/issues/82#issuecomment-1555306466)

OK, that fix works.

* Still I have to restart the app after changing the server to my personal one. The app doesn't tell me so I have to guess.
* Also to me it's still not clear wether I am connected or not. No tooltip on the "connection" icon, no message from the application. I suppose I am connected when the wheel stops spinning. After a few minutes it seems to lose connection (icon disappears?). Tested aside with Sparrow where I get feedback and stay connected.
* Next problem: No connection of the device possible, neither with USB nor with BLE. Terminal (lsusb) tells me: `Bus 003 Device 005: ID 1a86:55d4 QinHeng Electronics USB Single Serial` so it is properly connected. The application doesn't show or tell me anything about it.

> > @amosgroth try [#82 (comment)](https://github.com/Blockstream/green_qt/issues/82#issuecomment-1555306466)
> 
> OK, that fix works.
> 
>     * Still I have to restart the app after changing the server to my personal one. The app doesn't tell me so I have to guess.
> 
>     * Also to me it's still not clear wether I am connected or not. No tooltip on the "connection" icon, no message from the application. I suppose I am connected when the wheel stops spinning. After a few minutes it seems to lose connection (icon disappears?). Tested aside with Sparrow where I get feedback and stay connected.
> 
>     * Next problem: No connection of the device possible, neither with USB nor with BLE. Terminal (lsusb) tells me: `Bus 003 Device 005: ID 1a86:55d4 QinHeng Electronics USB Single Serial` so it is properly connected. The application doesn't show or tell me anything about it.

Helped myself with https://help.blockstream.com/hc/en-us/articles/900005443223-Why-is-Blockstream-Jade-not-recognized-by-my-computer-

Meaning I could upgrade the firmware but still the application doesn't connect and I cannot see the balance or anything, it stays like this:

![Capture d’écran du 2023-06-02 06-57-36](https://github.com/Blockstream/green_qt/assets/2759719/5548ffba-bd6d-451a-83aa-f17797cc2e0e)

**Comment by borgbyte** [comment]
@amosgroth 

> the application slows down the whole system, showing the spinning wheel ball and needs to be force-quitted.

please try version 1.2.3.

> Via BLE I could connect with system tools but the application doesn't recognize the device:

the app does not support BLE yet.

> my firewall noticed initial communication attempts of green to google servers,

some stuff is hosted on google buckets, like software update info and blog content.

**Comment by amosgroth** [comment]
> @amosgroth
> 
> > the application slows down the whole system, showing the spinning wheel ball and needs to be force-quitted.
> 
> please try version 1.2.3.
> 

The application in Linux recognizes and shows the device and I could change the firmware. But the device stays "locked" and there is no prompt (neither on the device nor in the application) to actually unlock it with the given password. So nothing has changed with this version....

> > Via BLE I could connect with system tools but the application doesn't recognize the device:
> 
> the app does not support BLE yet.
> 
> > my firewall noticed initial communication attempts of green to google servers,
> 
> some stuff is hosted on google buckets, like software update info and blog content.

As I said, I could imagine that I am not the only one that doesn't like it (the google servers connection).

**Comment by ardier16** [comment]
@amosgroth Thanks for the detailed follow-up, and apologies again for the delayed initial response.

The original issue reported here (entering a crash loop after enabling the proxy) was fixed in [Blockstream/gdk@c86946a](https://github.com/Blockstream/gdk/commit/c86946a8079067561b4027f93f6db3b3fe33a0fb) and is included in current app releases. Your later testing also confirmed that the app could enable Tor without reproducing the original crash loop.

The subsequent comments cover several separate problems, including custom Electrum connectivity, connection-status feedback, Linux AppImage startup, and Jade USB/Bluetooth discovery. These have different causes and should be tracked independently.

We’ll close this issue as completed. If any of those other problems still reproduce with the [latest release](https://github.com/Blockstream/green_qt/releases/latest), please open a separate issue for each one with the app version, operating system, configuration, and relevant logs so they can be investigated individually.
